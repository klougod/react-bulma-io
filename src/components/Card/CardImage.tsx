import React from 'react'
import PropTypes from 'prop-types'
import Figure from '../Figure'

export const CardImage = ({ className, alt, src, children }: any) => (
  <div className='card-image'>
    {src ? <Figure src={src} alt={alt} className={className} /> : { children }}
  </div>
)

CardImage.defaultProps = {
  className: '',
  alt: '',
  src: ''
}

CardImage.propTypes = {
  className: PropTypes.string,
  alt: PropTypes.string,
  src: PropTypes.string,
  children: PropTypes.node
}

export default CardImage
