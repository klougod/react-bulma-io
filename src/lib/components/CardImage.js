import React from 'react'
import PropTypes from 'prop-types'
import Figure from './Figure'

export const CardImage = props => {

  const {
    className,
    alt,
    src
  } = props

  return (
    <div className="card-image">
      <Figure src={src} alt={alt} className={className} />
    </div>
  )
}

CardImage.defaultProps = {
  className: '',
  alt: '',
  src: ''
}

CardImage.propTypes = {
  className: PropTypes.string,
  alt: PropTypes.string,
  src: PropTypes.string
}

export default CardImage