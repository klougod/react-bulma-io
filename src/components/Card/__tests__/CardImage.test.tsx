import React from 'react'
import Enzyme from 'enzyme'

import CardImage from '../CardImage'

describe('CardImage component test', function () {
  it('should render without crash without props', function () {
    const cardImageMock = Enzyme.shallow(<CardImage />)
    expect(cardImageMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const cardImageMockProps = {
      className: 'cardImageCLass',
      src: 'imageSrcUrl',
      alt: 'image alt'
    }
    const cardImageMock = Enzyme.shallow(<CardImage {...cardImageMockProps} />)
    expect(cardImageMock).toMatchSnapshot()
  })
})
