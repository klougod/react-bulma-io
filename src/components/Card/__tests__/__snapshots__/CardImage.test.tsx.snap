// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage component test should render without crash with props 1`] = `
<div
  className="card-image"
>
  <Component
    alt="image alt"
    className="cardImageCLass"
    src="imageSrcUrl"
  />
</div>
`;

exports[`CardImage component test should render without crash without props 1`] = `
<div
  className="card-image"
>
  <Component />
</div>
`;
